jsonnet/config.libsonnet

```diff
@@ -2,6 +2,7 @@
   // import kube-prometheus and override configurations
   kp:
     (import 'kube-prometheus/main.libsonnet') +
+    (import 'kube-prometheus/addons/all-namespaces.libsonnet') +
```

Comment:
What's the limitation with main.libsonnet?

Reply:
Prometheus is not allowed to watch services, pods and endpoints in namespaces other than kube-system, default and the namespace in which it is getting deployed. Due to which service discovery and metrics collection were not working as expected.

With `kube-prometheus/addons/all-namespaces.libsonnet` clusterrole will include extra roles.